### PR TITLE
Add parallelism to lighting queue

### PR DESCRIPTION
**Pull request by hafarooki**
(no i dont actually expect this to get merged)

### CODE REVIEW

File: Spigot-Server-Patches/0028-Lighting-Queue.patch [hunk 1]
```diff
@@ -136,41 +136,49 @@ index e8c3dcc24..c47f56ed8 100644
...
 +
++    private static ForkJoinPool pool = (ForkJoinPool) Executors.newWorkStealingPool();
 +
 +    static void processQueue(long curTime) {
 +        updatesThisTick = 0;
 +
 +        final long startTime = System.nanoTime();
 +        final long maxTickTime = MAX_TIME - (startTime - curTime);
 +
++
```
Comment (hafarooki) [inline Spigot-Server-Patches/0028-Lighting-Queue.patch]: why is this even here

File: Spigot-Server-Patches/0028-Lighting-Queue.patch [hunk 2]
```diff
@@ -136,41 +136,49 @@ index e8c3dcc24..c47f56ed8 100644
...
 +        START:
 +        for (World world : MinecraftServer.getServer().getWorlds()) {
 +            if (!world.paperConfig.queueLightUpdates) {
 +                continue;
 +            }
 +
 +            ObjectCollection<Chunk> loadedChunks = ((WorldServer) world).getChunkProviderServer().chunks.values();
-+            for (Chunk chunk : loadedChunks.toArray(new Chunk[loadedChunks.size()])) {
```
Comment (hafarooki) [inline Spigot-Server-Patches/0028-Lighting-Queue.patch]: idk why this was there so i removed it

File: Spigot-Server-Patches/0028-Lighting-Queue.patch [hunk 3]
```diff
@@ -136,41 +136,49 @@ index e8c3dcc24..c47f56ed8 100644
...
-+                }
++            for (Chunk chunk : loadedChunks) {
++                if (chunk.lightingQueue.isEmpty()) continue;
++                pool.submit(() -> {
++                    chunk.lightingQueue.processQueue(startTime, maxTickTime);
++                });
 +            }
 +        }
++
++        pool.awaitQuiescence(1000, TimeUnit.MILLISECONDS);
```
Comment (hafarooki) [inline Spigot-Server-Patches/0028-Lighting-Queue.patch:180]: not sure if this is the best way to do this

File: Spigot-Server-Patches/0028-Lighting-Queue.patch [hunk 4]
```diff
@@ -136,41 +136,49 @@ index e8c3dcc24..c47f56ed8 100644
...
 +
 +            ObjectCollection<Chunk> loadedChunks = ((WorldServer) world).getChunkProviderServer().chunks.values();
-+            for (Chunk chunk : loadedChunks.toArray(new Chunk[loadedChunks.size()])) {
-+                if (chunk.lightingQueue.processQueue(startTime, maxTickTime)) {
-+                    break START;
-+                }
++            for (Chunk chunk : loadedChunks) {
++                if (chunk.lightingQueue.isEmpty()) continue;
++                pool.submit(() -> {
++                    chunk.lightingQueue.processQueue(startTime, maxTickTime);
```
Comment (hafarooki) [inline Spigot-Server-Patches/0028-Lighting-Queue.patch]: lol this is probably pure stupid

File: Spigot-Server-Patches/0028-Lighting-Queue.patch (hunk 1 above)
Comment (aikar) [reply Spigot-Server-Patches/0028-Lighting-Queue.patch]: ... not a great way to inspire confidence you know what you are doing with this PR to open with that.

File: Spigot-Server-Patches/0028-Lighting-Queue.patch (hunk 2 above)
Comment (aikar) [reply Spigot-Server-Patches/0028-Lighting-Queue.patch]: again..... if you don't understand why this is happening....

File: Spigot-Server-Patches/0028-Lighting-Queue.patch (hunk 1 above)
Comment (hafarooki) [reply Spigot-Server-Patches/0028-Lighting-Queue.patch]: pls am sorry was just extra line no clu why add

File: Spigot-Server-Patches/0028-Lighting-Queue.patch (hunk 2 above)
Comment (hafarooki) [reply Spigot-Server-Patches/0028-Lighting-Queue.patch]: you dont need to specify the size in the type reference array yknow

**Comment by aikar** [comment]
please do not use such immature and offensive branch names like this.
But as discussed, this is not a clean solution, and has concurrency risks with chunk loads.

This is a task best saved for my work on ParaTask outside of cat finishing his branch.

**Pull request closed without merging**